src/collections/_documentation/development/sdk-dev/unified-api.md

```diff
@@ -54,7 +54,15 @@ meant that certain integrations (such as breadcrumbs) were often not possible.
 
 - **integration**: Code that provides middlewares, bindings or hooks into certain frameworks or environments, along with code that inserts those bindings and activates them. Usage for integrations does not follow a common interface.
 
-- **event processors:** Are callbacks that run for every event. They can either return a "new" event which in most cases means just adding data OR return `null` in case the event will be dropped and not sent.
+- **event processors**: Are callbacks that run for every event. They can either return a "new" event which in most cases means just adding data OR return `null` in case the event will be dropped and not sent.
+
+- **disabled SDK**: Before `init` has been called, or when no client is active,
```

Comment:
There are two disabled states: Having no client, or having a client without a transport. One can reach both states by either not calling init, or by calling init with an empty string as DSN.

If you call `init("")` you get breadcrumbs recorded.

Reply:
> One can reach both states by either not calling init, or by calling init with an empty string as DSN.

as you can see below, the event pipeline should run in this case. IMO its a valid usecase to have no DSN when you are using a custom transport.

Reply:
In Android (/cc @marandaneto )and .NET it's all NoOp until init is called. If init is called with an empty DSN, nothing happens but it sounds it's the only SDK doing that.

Reply:
actually, if you create a `Hub` by yourself with a `null` or `empty` DSN you even get `IllegalArgumentException` on Android.
If you create a `SentryClient` with a `null` or `empty` DSN and no transport is set, its also throwing `IllegalArgumentException`
Only thru the Sentry static class with an empty/null DSN that becomes NoOp for everything on Android.
Yes, it seems to be different than others.

Reply:
`null` IMHO should throw. The spec says empty string.

Throwing on initialization isn't that big deal since it should repro during the app start and during tests, shows the misconfiguration.

Something that is completely against the guideline for any public method **after** the SDK is initialized, which should never throw.

`Hub` and `SentryClient` are much simpler if can only be properly instantiated if a valid DSN is provided. So the guard for `""` happens outside (`Sentry.init` does that). The documentation never made clear that "" as a DSN in any code path should mean empty string and for that  reason that guideline was taken only at the public documented API `Sentry.init`. That said, it was by design.